osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableSwell.cs

```diff
@@ -231,24 +145,24 @@ protected override void UpdateStartTimeStateTransforms()
         {
             base.UpdateStartTimeStateTransforms();
 
-            using (BeginDelayedSequence(-ring_appear_offset))
-                targetRing.ScaleTo(target_ring_scale, 400, Easing.OutQuint);
+            (spinnerBody.Drawable as ISkinnableSwell)?.ApplyPassiveTransforms(this, MainPiece);
         }
 
         protected override void UpdateHitStateTransforms(ArmedState state)
         {
-            const double transition_duration = 300;
-
             switch (state)
             {
                 case ArmedState.Idle:
-                    expandingRing.FadeTo(0);
+                    HandleUserInput = true;
```

Comment:
What's with all of this `HandleUserInput` state management? This looks like it's actually changing gameplay logic for some reason? Explanation would be appreciated.

Reply:
The general reason why this is necessary is because `DrawableSwell`'s life is extended to make room for completion animations. Without stopping the handling of user inputs when `DrawableSwell`'s hitobject is completed, there will be an overlap with hit objects that come after making them impossible to hit. I added a test specifically for this behavior as I agree it might be a bit strange.

This specific case on `Idle` state is only there for rewind support. When the hitobject is rewound from being completed/missed to being idle, we want to make sure it accepts user inputs again.